File: test_api.mjs
// Use native fetch

async function test() {
    const headers = { 'Connect-Protocol-Version': '1', 'Content-Type': 'application/json' };

    // 1. Get an item
    console.log('Fetching items...');
    const res1 = await fetch('https://api.deltaforceapi.com/deltaforceapi.gateway.v1.ApiService/ListAuctionItems', {
        method: 'POST',
        headers,
        body: JSON.stringify({ pageSize: 1 })
    });
    const data1 = await res1.json();
    const item = data1.items[0];
    console.log('Got item:', item.name, item.id);

    // 2. Test GetAuctionItemPrices
    console.log('Testing GetAuctionItemPrices...');
    const now = new Date();
    const thirtyDaysAgo = new Date(now.getTime() - 30 * 24 * 60 * 60 * 1000);

    try {
        const res2 = await fetch('https://api.deltaforceapi.com/deltaforceapi.gateway.v1.ApiService/GetAuctionItemPrices', {
            method: 'POST',
            headers,
            body: JSON.stringify({
                auctionItemId: item.id,
                pageSize: 10,
                startTime: thirtyDaysAgo.toISOString(),
                endTime: now.toISOString()
            })
        });
        console.log('Prices status:', res2.status);
        const data2 = await res2.json();
        console.log('Prices data:', JSON.stringify(data2).substring(0, 200));
    } catch (e) { console.error('Error:', e.message); }

    // 3. Test GetAuctionItemPriceSeries
    console.log('Testing GetAuctionItemPriceSeries...');
    try {
        const res3 = await fetch('https://api.deltaforceapi.com/deltaforceapi.gateway.v1.ApiService/GetAuctionItemPriceSeries', {
            method: 'POST',
            headers,
            body: JSON.stringify({
                auctionItemId: item.id,
                startTime: thirtyDaysAgo.toISOString(),
                endTime: now.toISOString(),
                interval: 'INTERVAL_DAY'
            })
        });
        console.log('Series status:', res3.status);
        const data3 = await res3.json();
        console.log('Series data keys:', Object.keys(data3));
        console.log('Series data:', JSON.stringify(data3).substring(0, 200));
    } catch (e) { console.error('Error:', e.message); }
}

test();

File: test_api.py
import urllib.request
import json
import datetime

url_base = 'https://api.deltaforceapi.com/deltaforceapi.gateway.v1.ApiService/'
headers = {'Connect-Protocol-Version': '1', 'Content-Type': 'application/json'}

def post(endpoint, body):
    req = urllib.request.Request(url_base + endpoint, data=json.dumps(body).encode('utf-8'), headers=headers)
    try:
        with urllib.request.urlopen(req) as r:
            result = r.read()
            return json.loads(result)
    except urllib.error.HTTPError as e:
        print(f"Error {endpoint}: {e.code} {e.reason}\n{e.read().decode('utf-8')}")
        return None

print("Fetching item...")
item_data = post('ListAuctionItems', {'pageSize': 1})
if item_data:
    item = item_data.get('items', [{}])[0]
    item_id = item.get('id')
    print(f"Item ID: {item_id}")
    
    thirty_days_ago = (datetime.datetime.utcnow() - datetime.timedelta(days=30)).isoformat() + "Z"
    now = datetime.datetime.utcnow().isoformat() + "Z"
    
    print("\nFetching historical prices...")
    price_data = post('GetAuctionItemPrices', {
        'auctionItemId': item_id,
        'pageSize': 10,
        'startTime': thirty_days_ago,
        'endTime': now
    })
    if price_data:
        print(f"Price response keys: {list(price_data.keys())}")
        print(f"First price entry: {price_data.get('items', [{}])[0]}")
    
    print("\nFetching price series...")
    series_data = post('GetAuctionItemPriceSeries', {
        'auctionItemId': item_id,
        'startTime': thirty_days_ago,
        'endTime': now,
        'interval': 'INTERVAL_DAY'
    })
    if series_data:
        print(f"Series response keys: {list(series_data.keys())}")
        print(f"First series entry: {series_data.get('series', [{}])[0]}")
